tests/e2e/basic_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite includes test cases to verify basic functionality
+// in most common configurations
+
+package e2e
+
+import (
+	"log"
+	"os"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+)
+
+type BasicTestSuite struct {
+	esxName       string
+	volumeName    string
+	containerName string
+}
+
+func (s *BasicTestSuite) SetUpTest(c *C) {
+	s.esxName = os.Getenv("ESX")
+	s.volumeName = inputparams.GetVolumeNameWithTimeStamp()
+	s.containerName = inputparams.GetContainerNameWithTimeStamp()
+}
+
+var _ = Suite(&BasicTestSuite{})
+
+// Test volume lifecycle management on different datastores:
+// VM1 - local VMFS datastore
+// VM2 - shared VMFS datastore
+// VM3 - shared VSAN datastore
+func (s *BasicTestSuite) TestVolumeCreation(c *C) {
```

Comment:
nit: better to add details steps carried from GH issue or confluence page to get a quick sense what the test method is trying to do instead of scanning each line.